native/src/widget/scrollable.rs

```diff
@@ -970,10 +1311,56 @@ impl Scrollbar {
         grabbed_at: f32,
         cursor_position: Point,
     ) -> f32 {
-        (cursor_position.y
-            - self.bounds.y
-            - self.scroller.bounds.height * grabbed_at)
-            / (self.bounds.height - self.scroller.bounds.height)
+        match self.direction {
+            Direction::Vertical => {
+                (cursor_position.y
+                    - self.bounds.y
+                    - self.scroller.bounds.height * grabbed_at)
+                    / (self.bounds.height - self.scroller.bounds.height)
+            }
+            Direction::Horizontal => {
+                (cursor_position.x
+                    - self.bounds.x
+                    - self.scroller.bounds.width * grabbed_at)
+                    / (self.bounds.width - self.scroller.bounds.width)
+            }
+        }
+    }
+}
+
+/// The directional offset of a [`Scrollable`].
+#[derive(Debug, Clone, Copy)]
+enum Offset {
+    Absolute(f32),
+    Relative(f32),
+}
+
+impl Offset {
+    fn absolute(
+        self,
+        direction: Direction,
+        bounds: Rectangle,
+        content_bounds: Rectangle,
+    ) -> f32 {
+        match self {
+            Self::Absolute(absolute) => match direction {
+                Direction::Horizontal => {
+                    absolute.min((content_bounds.width - bounds.width).max(0.0))
+                }
+                Direction::Vertical => absolute
+                    .min((content_bounds.height - bounds.height).max(0.0)),
+            },
+            Self::Relative(percentage) => match direction {
+                Direction::Horizontal => {
+                    ((content_bounds.width - bounds.width) * percentage)
+                        .max(0.0)
+                }
+                Direction::Vertical => {
+                    ((content_bounds.height - bounds.height) * percentage)
+                        .max(0.0)
+                }
+            },
+        }
```

Comment:
I think we can just split this into `absolute_x` and `absolute_y` and just call the method accordingly, since we never have a dynamic `Direction`.